src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1388,6 +1523,30 @@ public static void Xml_TypeWithArrayPropertyHavingChoice()
         Assert.NotNull(actual.ManyChoices);
         Assert.Equal(value.ManyChoices.Length, actual.ManyChoices.Length);
         Assert.True(Enumerable.SequenceEqual(value.ManyChoices, actual.ManyChoices));
+
+        // Try again with a null array
+        value = new TypeWithArrayPropertyHavingChoice() { ManyChoices = null, ChoiceArray = itemChoices };
+        actual = SerializeAndDeserialize(value, WithXmlHeader("<TypeWithArrayPropertyHavingChoice xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" />"));
+        Assert.NotNull(actual);
```

Comment:
What happened to ChoiceArray in the XML? It looks like it should have 2 items in it, "Item" and "Amount".

Reply:
Nevermind, I went and looked at the type definition.